liferay-frontend/liferay-portal
# LPD-28456 Reorganize frontend-data-set-view-web pw tests in preparation for easy migration

In this PR we are:

- Splitting Data Set Manager family of tests into 2 broad categories
  - `data-set-admin` covers the Data Set Manager UI provides all the features to create Data Sets
  - `data-set-fragment` covers the fragment renders according to the specification for a given Data Set
- Fixing some tests that were failing or at least flaky
- Making some optimizations related to
  - Test steps reusability
  - Common logic for `beforeEach()`
  - Avoiding dependency on data relationship names in test code, in preparation for data model renaming @markocikos 
  - Other minor polishing

This should allow separate evolution of both categories in terms of number of tests, file splitting, etc.

This is how final outcome looks like:
![image](https://github.com/liferay-frontend/liferay-portal/assets/861014/9ac8911a-7f23-4958-b878-cd15a50a9300)

This impacts on the ongoing work by @chestofwonders @thektan @kresimir-coko @markocikos and probably @juanjofgliferay 

Additional notes:

- Reference: https://liferay.atlassian.net/browse/LPD-28456
- Though ALL tests are green locally, I've detected some flakiness in a couple of tests. 
- I'm prioritizing merging this big chunk of code over stabilizing tests as there is a lot of blockers we want to remove.
- Once @markocikos finishes with the renaming of the app we might tweak some other file naming accordingly, but we need to merge all this before
